transcendence-app/src/auth/oauth42.strategy.ts

```diff
@@ -25,13 +28,30 @@ export class OAuth42Strategy extends PassportStrategy(Strategy, 'oauth42') {
   }
 
   async validate(accessToken: string): Promise<User | null> {
-    const userDto = await this.api42Service.get42UserData(accessToken);
+    const { userDto, avatarUrl } = await this.api42Service.get42UserData(
+      accessToken,
+    );
     const dbUser = await this.userService.retrieveUserWithUserName(
       userDto.username,
     );
     if (dbUser) {
       return dbUser;
     }
-    return this.userService.addUser(userDto);
+
+    // Download the avatar and save the avatar metadata and the user in the database
```

Comment:
Couldn't we make one or two private, properly named, methods in this OAuth42Strategy class for these avatar file operations, instead of just leaving some comments?

Reply:
Done, :+1: